### PR TITLE
Update mailman configuration to match latest version of mailman on 20.04 [BB-4467]

### PR DESCRIPTION
**Pull request by xitij2000**
Update mailman3 configuration to work with the latest release of mailman3 on Ubuntu 20.04.

### CODE REVIEW

File: playbooks/roles/mail/templates/mailman-web.py.j2 [hunk 1]
```diff
@@ -32,7 +32,6 @@ INSTALLED_APPS = (
     'django.contrib.staticfiles',
     'rest_framework',
     'django_gravatar',
-    'paintstore',
```
Comment (xitij2000) [inline playbooks/roles/mail/templates/mailman-web.py.j2:35]: This is no longer used with the newer version of mailmain3

File: playbooks/roles/mail/templates/mailman-web.py.j2 [hunk 2]
```diff
@@ -72,3 +71,4 @@ USE_TZ = True
 
 DEFAULT_FROM_EMAIL = '{{ MAILMAN_FROM_EMAIL }}'
 SERVER_EMAIL = '{{ MAILMAN_FROM_EMAIL }}'
+POSTORIUS_TEMPLATE_BASE_URL = 'https://{{ MAILMAN_WEB_DOMAIN }}'
```
Comment (xitij2000) [inline playbooks/roles/mail/templates/mailman-web.py.j2:74]: This is a new configuration value.

File: playbooks/roles/mail/templates/mailman3-nginx-proxy.conf.j2 [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # {{ ansible_managed }}
 
 upstream mailman3 {
-    server unix:/run/mailman3/web/uwsgi.sock fail_timeout=0;
+    server unix:/run/mailman3-web/uwsgi.sock fail_timeout=0;
```
Comment (xitij2000) [inline playbooks/roles/mail/templates/mailman3-nginx-proxy.conf.j2:4]: The location of this socket has changed in either ubunutu's config or the upsteream itself.

**Comment by swalladge** [comment]
@xitij2000 while you're here, could you also remove the [block device mount](https://github.com/open-craft/ansible-playbooks/blob/a2ec8d9fd1bce40db44f05fdd8b7505d71f19a48/playbooks/roles/mail/tasks/main.yml#L3-L9)?  We no longer use it, since the latest redeployment.  CC @giovannicimolin

**Comment by xitij2000** [comment]
> @xitij2000 while you're here, could you also remove the [block device mount](https://github.com/open-craft/ansible-playbooks/blob/a2ec8d9fd1bce40db44f05fdd8b7505d71f19a48/playbooks/roles/mail/tasks/main.yml#L3-L9)? We no longer use it, since the latest redeployment. CC @giovannicimolin

Done.

**Review by 0x29a — APPROVED** [review]
:+1:

- [x] I tested that mail server is working fine after the upgrade.
- [x] I read through the diff.

**Pull request merged**